File: tx-rollback-test/src/main/java/com/example/txrollbacktest/post/application/PostRegister.java
package com.example.txrollbacktest.post.application;

import com.example.txrollbacktest.post.domain.Post;
import com.example.txrollbacktest.post.port.PersistPostPort;
import org.springframework.stereotype.Component;

@Component
final class PostRegister {

    private final PersistPostPort persistPostPort;

    PostRegister(PersistPostPort persistPostPort) {
        this.persistPostPort = persistPostPort;
    }

    public Post add(String title, String content) {
        return persistPostPort.save(Post.of(title, content));
    }
}

File: cinema-friends/domain/src/main/kotlin/com/cinemafriends/post/application/service/PostRegister.kt
package com.cinemafriends.post.application.service

import com.cinemafriends.post.application.port.output.PersistPostPort
import com.cinemafriends.post.entity.Post
import org.springframework.stereotype.Component

@Component
class PostRegister(
    private val persistPostPort: PersistPostPort
) {
    fun save(post: Post): Long {
        val post = persistPostPort.persist(post)
        return post.id
    }
}